nbsmoke/__init__.py

```diff
@@ -162,4 +162,8 @@ def pytest_collect_file(path, parent):
                 dowhat = LintNb
             elif opt.nbsmoke_verify:
                 dowhat = VerifyNb
+
+            if hasattr(IPyNbFile, "from_parent"):
+                return IPyNbFile.from_parent(parent, fspath=path)
```

Comment:
Does it need to be something like this? 

```suggestion
                x = IPyNbFile.from_parent(parent, fspath=path)
                x._dowhat = dowhat
                return x
```

I'm just guessing based on the docs linked from the last build on master:

```python
==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py:337: in traced_hookexec
    return outcome.get_result()
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py:335: in <lambda>
    outcome = _Result.from_call(lambda: oldcall(hook, hook_impls, kwargs))
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/nbsmoke/__init__.py:165: in pytest_collect_file
    return IPyNbFile(path, parent, dowhat=dowhat)
/home/travis/build/pyviz-dev/nbsmoke/.tox/py37/lib/python3.7/site-packages/_pytest/nodes.py:95: in __call__
    warnings.warn(NODE_USE_FROM_PARENT.format(name=self.__name__), stacklevel=2)
E   pytest.PytestDeprecationWarning: Direct construction of IPyNbFile has been deprecated, please use IPyNbFile.from_parent.
E   See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent for more details.
=========================== short test summary info ============================
ERROR  - pytest.PytestDeprecationWarning: Direct construction of IPyNbFile ha...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.21s ===============================
=========================== short test summary info ============================
FAILED tests/test_lint.py::test_lint_bad_silenced_with_noqa - assert <ExitCod...
======================== 1 failed, 37 passed in 10.81s =========================
```

Reply:
good looks. i think you can use dowhat in the kwargs. i think that should do the trick.